server/src/main/java/org/elasticsearch/cluster/metadata/MetaDataIndexStateService.java

```diff
@@ -121,28 +210,174 @@ public ClusterState closeIndices(ClusterState currentState, final Index[] indice
         RestoreService.checkIndexClosing(currentState, indicesToClose);
         // Check if index closing conflicts with any running snapshots
         SnapshotsService.checkIndexClosing(currentState, indicesToClose);
-        logger.info("closing indices [{}]", indicesAsString);
-
-        MetaData.Builder mdBuilder = MetaData.builder(currentState.metaData());
-        ClusterBlocks.Builder blocksBuilder = ClusterBlocks.builder()
-            .blocks(currentState.blocks());
-        for (IndexMetaData openIndexMetadata : indicesToClose) {
-            final String indexName = openIndexMetadata.getIndex().getName();
-            mdBuilder.put(IndexMetaData.builder(openIndexMetadata).state(IndexMetaData.State.CLOSE));
-            blocksBuilder.addIndexBlock(indexName, INDEX_CLOSED_BLOCK);
+
+        // If the cluster is in a mixed version that does not support the shard close action,
+        // we use the previous way to close indices and directly close them without sanity checks
+        final boolean useDirectClose = currentState.nodes().getMinNodeVersion().before(Version.V_7_0_0);
```

Comment:
are there mixed version cluster tests that check index closing?

Reply:
The :qa:mixed-cluster tests runs the core REST tests and those tests contains some open/close YAML tests that fail without this logic